foundation/buf.lock

```diff
@@ -0,0 +1,7 @@
+# Generated by buf. DO NOT EDIT.
```

Comment:
I believe the `foundation` was the repo/module name that was agreed to go into, https://buf.build/mobilecoin
unless we create a new `mobilecoinfoundation` owner